Boltdir/site-modules/metrics/files/collect_metrics_files.rb

```diff
@@ -0,0 +1,377 @@
+    #
+    # @return [void]
+    #
+    # @example
+    #   inspect_metrics_dir_for_service_dirs
+    def inspect_metrics_dir_for_service_dirs
+      metrics_dir_obj = Dir.open @metrics_dir
+      metrics_dir_obj.each do |path|
+        next if path =~ /^scripts$|^bin$|^\.$|^\.\.$/
+
+        absolute_path = "#{metrics_dir_obj.path}/#{path}"
+        service = path
+        next unless File.directory? absolute_path
+
+        inspect_service_dir_for_metrics_files(absolute_path, service)
+      end
+    end
+
+    # Process potential service dirs or subdirs
+    #
+    # Uses recursion but doesn't care which subdir the files come from, they all go in the
+    # same output service dir
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] dir_path Service dir or subdir
+    # @param [string] service name of the service the directory belongs to
+    #
+    # @return [void]
+    #
+    # @example
+    #   inspect_service_dir_for_metrics_files(dir_path, service)
+    def inspect_service_dir_for_metrics_files(dir_path, service)
+      service_dir = Dir.open dir_path
+      service_dir.each do |path|
+        next if path =~ /^\.$|^\.\.$/
+
+        absolute_path = "#{service_dir.path}/#{path}"
+        if File.directory?(absolute_path)
+          inspect_service_dir_for_metrics_files(absolute_path, service)
+          next
+        end
+        if path =~ /.json$/
+          consider_json_for_inclusion(absolute_path)
+        elsif path =~ /.tar.gz$/
+          consider_tarfile_contents_for_inclusion(absolute_path, service)
+        end
+      end
+    end
+
+    # Determines if a json file is from the target window
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] filename json filename
+    #
+    # @return [boolean] true if yes, false if no
+    #
+    # @example
+    #   json_file_from_target_window(filename)
+    def json_file_from_target_window?(filename)
+      time_str = filename.gsub(/.json$/, "")
+      time_obj = DateTime.parse(time_str)
+      epoch = time_obj.to_time.to_i
+      return true if epoch > @start_epoch && (epoch - @poll_interval) < @end_epoch
+
+      return false
+    end
+
+    # Determines if a json file is from the target window and if so, stages it to the staging_dir
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] file_path absolute path to json file
+    #
+    # @return [void]
+    #
+    # @example
+    #   consider_json_for_inclusion(file_path)
+    def consider_json_for_inclusion(file_path)
+      filename = File.basename(file_path)
+      stage_json(file_path) if json_file_from_target_window?(filename)
+    end
+
+    # Stage json to the output_dir
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] file_path absolute path to json file
+    #
+    # @return [void]
+    #
+    # @example
+    #   stage_json(file_path)
+    def stage_json(file_path)
+      filename = File.basename(file_path)
+      file_dir = File.dirname(file_path)
```

Comment:
FYI, the [split method](https://ruby-doc.org/core-2.6.3/File.html#method-c-split) will do this in one call.
```
file_dir, filename = File.split(file_path)
```